crates/storage/libmdbx-rs/src/environment.rs

```diff
@@ -589,6 +676,40 @@ impl EnvironmentBuilder {
                     ))?;
                 }
 
+                if let Some(handle_slow_readers) = self.handle_slow_readers {
+                    let handle_slow_readers = Box::leak(Box::new(handle_slow_readers));
+                    let hsr = Box::leak(Box::new(
+                        |_env: *const MDBX_env,
+                         _txn: *const MDBX_txn,
+                         pid: mdbx_pid_t,
+                         tid: mdbx_tid_t,
+                         laggard: u64,
+                         gap: ::libc::c_uint,
+                         space: usize,
+                         retry: ::libc::c_int|
+                         -> i32 {
+                            handle_slow_readers(
+                                pid as u32,
+                                tid as u32,
+                                laggard,
+                                gap as usize,
+                                space,
+                                retry as isize,
+                            )
+                            .into()
+                        },
+                    ));
```

Comment:
can we extract this to a new function?

Reply:
agreed, also added a test to be sure FFI works as expected with no segfaults